### PR TITLE
Simulation logic

### LINKED ISSUE #50: The visible part of simulation (20)
- [x] implement simple layout (showing the relevant components)
- [ ] Add functionality to highlight locations/transitions (highlight det der sendes tilbage med test response)

### LINKED ISSUE #84: Add functionality to highlight transitions based on SimulationResponse (5)
- [x] consider edge cases (robusthed, hvornår kan den fejle?)
- [ ] documentation in code
- [ ] test

### LINKED ISSUE #85: Add functionality to highlight locations based on Simulation Response (3)
- [ ] consider edge cases (robusthed, hvornår kan den fejle?)
- [ ] documentation in code
- [ ] test

### LINKED ISSUE #86: Send SimulationStartRequest to backend when requested by user (2)
- [x] clean up of controller used to call this (i SimulationController)
- [ ] inline documentation/comments
- [x] remove demo data (backend will provide engine with test data)

### LINKED ISSUE #88: Send SimulationStepRequest to backend when requested by user
- [ ] remove demo data

### PR DESCRIPTION
**Pull request by EmilieSonne**
Closes #50, closes #84, closes #85, closes #86, closes #88.

### CODE REVIEW

File: src/main/java/ecdar/abstractions/Edge.java [hunk 1]
```diff
@@ -210,7 +210,7 @@ public void deserialize(final JsonObject json, final Component component) {
 
         final JsonPrimitive IDJson = json.getAsJsonPrimitive(ID);
         if (IDJson != null) setId(IDJson.getAsString());
```
Comment (WassawRoki) [inline src/main/java/ecdar/abstractions/Edge.java:212]: Så vi køre setId 2 gange hvis IDJson !=

Comment (APaludan) [reply src/main/java/ecdar/abstractions/Edge.java]: fixed. slettede det så den altid genererer nyt id

Comment (APaludan) [reply src/main/java/ecdar/abstractions/Edge.java]: fordi gamle projekter kan have forkert id så laver vi nyt hver gang vi loader projektet

Comment (APaludan) [reply src/main/java/ecdar/abstractions/Edge.java]: fixed igen igen. Nu laver den kun nyt id hvis IDJson ikke er unikt

File: src/main/java/ecdar/controllers/SimulatorController.java [hunk 2]
```diff
@@ -139,7 +123,7 @@ public void willHide() {
         overviewPresentation.getController().getComponentObservableList().forEach(component -> {
             // Previously reset coordinates of component box
         });
-        overviewPresentation.getController().unhighlightProcesses();
+        // overviewPresentation.getController().unhighlightProcesses();
```
Comment (Mati-AAU) [inline src/main/java/ecdar/controllers/SimulatorController.java:126]: why is this a comment?

File: src/main/java/ecdar/controllers/TransitionPaneElementController.java [hunk 3]
```diff
@@ -45,16 +43,16 @@ public class TransitionPaneElementController implements Initializable {
 
     @Override
     public void initialize(URL location, ResourceBundle resources) {
-        Ecdar.getSimulationHandler().availableTransitions.addListener((ListChangeListener<Transition>) c -> {
-            while (c.next()) {
-                for (Transition trans : c.getAddedSubList()) insertTransition(trans);
+        // Ecdar.getSimulationHandler().availableTransitions.addListener((ListChangeListener<Transition>) c -> {
+        //     while (c.next()) {
+        //         for (Transition trans : c.getAddedSubList()) insertTransition(trans);
```
Comment (Mati-AAU) [inline src/main/java/ecdar/controllers/TransitionPaneElementController.java:48]: why is this commented out?

File: src/main/java/ecdar/controllers/TransitionPaneElementController.java [hunk 4]
```diff
@@ -45,16 +43,16 @@ public class TransitionPaneElementController implements Initializable {
...
-                    transitionPresentationMap.remove(trans);
-                }
-            }
-        });
+        //         for (final Transition trans: c.getRemoved()) {
+        //             transitionList.getChildren().remove(transitionPresentationMap.get(trans));
+        //             transitionPresentationMap.remove(trans);
+        //         }
+        //     }
+        // });
```
Comment (Mati-AAU) [inline src/main/java/ecdar/controllers/TransitionPaneElementController.java:55]: why is this commented out?

File: src/main/java/ecdar/controllers/TransitionPaneElementController.java [hunk 5]
```diff
@@ -137,26 +135,25 @@ private void insertTransition(Transition transition) {
...
         // e.g. TransitionPresentation already has mouseEntered functionality and we want to keep it
         EventHandler mouseEntered = transitionPresentation.getOnMouseEntered();
-        transitionPresentation.setOnMouseEntered(event -> {
-            SimulatorController.setSelectedTransition(transitionPresentation.getController().getTransition());
-            mouseEntered.handle(event);
-        });
+        // transitionPresentation.setOnMouseEntered(event -> {
+        //     SimulatorController.setSelectedTransition(transitionPresentation.getController().getTransition());
+        //     mouseEntered.handle(event);
+        // });
```
Comment (Mati-AAU) [inline src/main/java/ecdar/controllers/TransitionPaneElementController.java:141]: comment?

File: src/main/java/ecdar/controllers/TransitionPaneElementController.java [hunk 6]
```diff
@@ -137,26 +135,25 @@ private void insertTransition(Transition transition) {
...
         EventHandler mouseExited = transitionPresentation.getOnMouseExited();
         transitionPresentation.setOnMouseExited(event -> {
-            SimulatorController.setSelectedTransition(null);
             mouseExited.handle(event);
         });
 
-        transitionPresentation.setOnMouseClicked(event -> {
-            event.consume();
+        // transitionPresentation.setOnMouseClicked(event -> {
+        //     event.consume();
```
Comment (Mati-AAU) [inline src/main/java/ecdar/controllers/TransitionPaneElementController.java:149]: comment?

File: src/main/java/ecdar/controllers/TransitionPaneElementController.java [hunk 7]
```diff
@@ -137,26 +135,25 @@ private void insertTransition(Transition transition) {
...
-            if (simHandler != null) {
-                simHandler.nextStep(transitionPresentation.getController().getTransition(), this.delay.get());
-            }
-        });
+        //     // Performs the next step of the simulation when clicking on a transition
+        //     SimulationHandler simHandler = Ecdar.getSimulationHandler();
+        //     if (simHandler != null) {
+        //         // simHandler.nextStep(transitionPresentation.getController().getTransition(), this.delay.get());
+        //     }
+        // });
```
Comment (Mati-AAU) [inline src/main/java/ecdar/controllers/TransitionPaneElementController.java:156]: comment?

File: src/main/java/ecdar/controllers/TransitionPaneElementController.java [hunk 8]
```diff
@@ -200,8 +197,7 @@ private void expandTransitions() {
      */
     @FXML
     private void refreshTransitions() {
-        SimulatorController.setSelectedTransition(null);
-//        MainController.openReloadSimulationDialog();         // ToDo: Implement
```
Comment (Mati-AAU) [inline src/main/java/ecdar/controllers/TransitionPaneElementController.java:204]: Er "ToDo: Implement" implementeret af det nye kode?

File: src/main/java/ecdar/simulation/SimulationState.java [hunk 9]
```diff
@@ -1,51 +1,42 @@
...
 import EcdarProtoBuf.ObjectProtos;
+import EcdarProtoBuf.ObjectProtos.State;
+import ecdar.Ecdar;
 import javafx.util.Pair;
 
-import java.math.BigDecimal;
 import java.util.ArrayList;
 
 public class SimulationState {
+    // key = component name, value = id
```
Comment (Mati-AAU) [inline src/main/java/ecdar/simulation/SimulationState.java:11]: comment?

File: src/main/java/ecdar/backend/SimulationHandler.java [hunk 10]
```diff
@@ -1,467 +1,347 @@
...
+//        }
+
+        // As the clocks values starts after the variables values in currentConcreteState.get().getVariables()
+        // Then i needs to start where the variables ends.
+        // j is needed to map the correct name with the value
+//        for (int i = noOfVars, j = 0; i < noOfClocks + noOfVars ; i++, j++) {
+//            simulationClocks.put(engine.getSystem().getClockName(j),
+//                    currentConcreteState.get().getVariables()[i].getValue(BigDecimal.ZERO));
+//        }
+    }
```
Comment (Mati-AAU) [inline src/main/java/ecdar/backend/SimulationHandler.java:247]: comment?

**Review by Mati-AAU — CHANGES_REQUESTED** [review]
The amount of deletions scare me but i assume its old unnecessary code

File: src/main/java/ecdar/simulation/SimulationState.java (hunk 9 above)
Comment (APaludan) [reply src/main/java/ecdar/simulation/SimulationState.java:11]: jaer det var for at forklare at de bliver gemt som et key-value pair hvor key repræsentarer component name og value er id på edge/location

File: src/main/java/ecdar/controllers/TransitionPaneElementController.java (hunk 8 above)
Comment (APaludan) [reply src/main/java/ecdar/controllers/TransitionPaneElementController.java:204]: Vi har ikke noget reloadSimulationDialog endnu så den restarter bare simulation. Har også omdøbt metoden til restartSimulation

File: src/main/java/ecdar/controllers/TransitionPaneElementController.java (hunk 3 above)
Comment (APaludan) [reply src/main/java/ecdar/controllers/TransitionPaneElementController.java]: slettet

File: src/main/java/ecdar/controllers/TransitionPaneElementController.java (hunk 4 above)
Comment (APaludan) [reply src/main/java/ecdar/controllers/TransitionPaneElementController.java]: slettet

File: src/main/java/ecdar/controllers/SimulatorController.java (hunk 2 above)
Comment (APaludan) [reply src/main/java/ecdar/controllers/SimulatorController.java:126]: slettet

File: src/main/java/ecdar/controllers/EcdarController.java [hunk 11]
```diff
@@ -726,7 +726,7 @@ private void updateScaling(double newScale) {
 
         scaleIcons(root, newCalculatedScale);
         editorPresentation.getController().scaleEdgeStatusToggle(newCalculatedScale);
-        messageTabPane.getController().updateScale(newScale);
+        /*messageTabPane.getController().updateScale(newScale);*/
```
[the remark this answers is not in the document]
Comment (APaludan) [reply src/main/java/ecdar/controllers/EcdarController.java:729]: slet

File: src/main/java/ecdar/controllers/EcdarController.java [hunk 12]
```diff
@@ -180,8 +180,8 @@ public void initialize(final URL location, final ResourceBundle resources) {
         initializeMenuBar();
         startBackgroundQueriesThread(); // Will terminate immediately if background queries are turned off
 
-        bottomFillerElement.heightProperty().bind(messageTabPane.maxHeightProperty());
-        messageTabPane.getController().setRunnableForOpeningAndClosingMessageTabPane(this::changeInsetsOfProjectAndQueryPanes);
+        /*bottomFillerElement.heightProperty().bind(messageTabPane.maxHeightProperty());*/
+        /*messageTabPane.getController().setRunnableForOpeningAndClosingMessageTabPane(this::changeInsetsOfProjectAndQueryPanes);*/
```
[the remark this answers is not in the document]
Comment (APaludan) [reply src/main/java/ecdar/controllers/EcdarController.java:184]: slet

File: src/main/java/ecdar/backend/SimulationHandler.java (hunk 10 above)
Comment (APaludan) [reply src/main/java/ecdar/backend/SimulationHandler.java:247]: skal måske bruges i fremtiden så vi tænkte at beholde det

File: src/main/java/ecdar/controllers/TransitionPaneElementController.java (hunk 5 above)
Comment (APaludan) [reply src/main/java/ecdar/controllers/TransitionPaneElementController.java:130]: skal måske bruges i fremtiden

File: src/main/java/ecdar/controllers/TransitionPaneElementController.java (hunk 6 above)
Comment (APaludan) [reply src/main/java/ecdar/controllers/TransitionPaneElementController.java:138]: skal måske bruges i fremtiden

File: src/main/java/ecdar/controllers/TransitionPaneElementController.java (hunk 4 above)
Comment (Mati-AAU) [reply src/main/java/ecdar/controllers/TransitionPaneElementController.java]: its still there, is it committed?

File: src/main/java/ecdar/controllers/TransitionPaneElementController.java (hunk 3 above)
Comment (Mati-AAU) [reply src/main/java/ecdar/controllers/TransitionPaneElementController.java]: its still there, is it committed?

File: src/main/java/ecdar/controllers/TransitionPaneElementController.java (hunk 8 above)
Comment (Mati-AAU) [reply src/main/java/ecdar/controllers/TransitionPaneElementController.java:204]: Seems the same, is it not committed?

Comment (APaludan) [reply src/main/java/ecdar/controllers/TransitionPaneElementController.java:204]: Ups ja har glemt at commit. Jeg tagger dig lige når det er færdigt så må du meget gerne kigge det igennem igen 😄

Comment (APaludan) [reply src/main/java/ecdar/controllers/TransitionPaneElementController.java:204]: ved ik hvorfor github ikke skriver "outdated" men den er altså ændret. Det ligger på linje 177 nu

**Comment by APaludan** [comment]
@Mati-AAU nu tror jeg det hele er fixed. Du må meget gerne kigge på det igen :)

File: src/main/java/ecdar/controllers/EcdarController.java (hunk 12 above)
[the remark this answers is not in the document]
Comment (Mati-AAU) [reply src/main/java/ecdar/controllers/EcdarController.java]: still there

File: src/main/java/ecdar/controllers/EcdarController.java (hunk 11 above)
[the remark this answers is not in the document]
Comment (Mati-AAU) [reply src/main/java/ecdar/controllers/EcdarController.java]: still there

File: src/main/java/ecdar/controllers/EcdarController.java (hunk 12 above)
[the remark this answers is not in the document]
Comment (APaludan) [reply src/main/java/ecdar/controllers/EcdarController.java]: billede
🤔

**Review by Mati-AAU — APPROVED** [review]

File: src/main/java/ecdar/controllers/EcdarController.java (hunk 12 above)
[the remark this answers is not in the document]
Comment (Mati-AAU) [reply src/main/java/ecdar/controllers/EcdarController.java]: github is being weird on my end, all good my bad

File: src/main/java/ecdar/controllers/EcdarController.java (hunk 11 above)
[the remark this answers is not in the document]
Comment (Mati-AAU) [reply src/main/java/ecdar/controllers/EcdarController.java]: github is being weird on my end, all good my bad

File: src/main/java/ecdar/controllers/TransitionPaneElementController.java (hunk 3 above)
Comment (Mati-AAU) [reply src/main/java/ecdar/controllers/TransitionPaneElementController.java]: github is being weird on my end, all good my bad

File: src/main/java/ecdar/controllers/TransitionPaneElementController.java (hunk 4 above)
Comment (Mati-AAU) [reply src/main/java/ecdar/controllers/TransitionPaneElementController.java]: github is being weird on my end, all good my bad

**Review by Archdragon79 — APPROVED** [review]
LGTM
